### PR TITLE
[Bagel] Support TeaCache

### PR DESCRIPTION
**Pull request by princepride**
## Purpose

Related: https://github.com/vllm-project/vllm-omni/issues/833 
This PR enables `tea_cache` acceleration for the **Bagel** model in `vllm-omni`.

1. Modified the model adaptation method in tea-cache to provide flexible support for models with different module names, similar to the implementation in cache-dit.

2. Currently, Coefficient directly utilizes FluxTransformer. These parameters will be updated accordingly once the Coefficient Estimation Tool is finalized.

## Test Plan

**Test Command:**
```
import torch
import time
from vllm_omni.entrypoints.omni_diffusion import OmniDiffusion

def run_test(cache_backend=None, cache_config=None, run_name="Baseline"):
    pipeline = OmniDiffusion(
        model="../models/BAGEL-7B-MoT",
        cache_backend=cache_backend,
        cache_config=cache_config
    )
    prompt = "A woman wear a blue dress"
    start_gen = time.time()
    result = pipeline.generate(
        prompt, 
        seed=52
    )
    end_gen = time.time()
    gen_time = end_gen - start_gen
    print(f"Generation Time: {gen_time:.4f}s")
    
    output_path = f"output_bagel_{run_name.lower().replace(' ', '_')}.png"
    result.images[0].save(output_path)

def main():
    # 1. Baseline (No Cache)
    baseline_time = run_test(cache_backend=None, cache_config=None, run_name="Baseline (No Cache)")
    
    cache_config = {
        # TeaCache parameters [tea_cache only]
        "rel_l1_thresh": 0.2,  # Threshold for accumulated relative L1 distance
        # Note: coefficients will use model-specific defaults based on model_type
        #       (e.g., QwenImagePipeline or FluxPipeline)
    }
    run_test(cache_backend="tea_cache", cache_config=cache_config, run_name="Tea Cache")

if __name__ == "__main__":
    main()

```
[Optional] You can directly use `examples/offline_inference/text_to_image/text_to_image.py`
```bash
python examples/offline_inference/text_to_image/text_to_image.py --model ../models/BAGEL-7B-MoT --cache_backend tea_cache
```

## Test Result

**Performance Comparison:**

| Metric | Baseline (No Cache) | Cache-Dit Enabled | Speedup |
| :--- | :--- | :--- | :--- |
| **Generation Time** | 9.35s | 4.36s | **~2.14x** |

**Visual Validation:**

<table style="width: 100%; border-collapse: collapse;">
  <tr>
    <td align="center" style="border: none;">
      <br/>
      <b>Baseline (No Cache)</b>
    </td>
    <td align="center" style="border: none;">
    <br/>
      <b>Cache-Dit Enabled</b>
    </td>
  </tr>
</table>

### CODE REVIEW

**Comment by princepride** [comment]
@SamitHuang I have calculated the polynomial coefficients using the tool https://github.com/vllm-project/vllm-omni/pull/940. However, since the tool relies on extract_context, it currently has a dependency on this PR. I would like to have this PR merged first.
